### PR TITLE
[WIP] (GH-300) Port Wyam.Hosting to netstandard2.0

### PR DESCRIPTION
**Pull request by RLittlesII**
Example Unit Tests
Wyam.Hosting.Tests
Wyam.Hosting.Tests.Owin
Live Reload

### CODE REVIEW

**Comment by MisinformedDNA** [comment]
Where are you with this?

**Comment by RLittlesII** [comment]
I am attempting to get current unit tests passing, and write unit tests for the new middleware that I created.

There are several concerns around Live Reload and how we are going to handle that.  But everything builds.

**Comment by MisinformedDNA** [comment]
What are your (and @daveaglick's) thoughts on Browser Sync? I don't have experience with either, but LiveReload is deprecated. I found this [nice comparison](https://www.slant.co/versus/5065/5066/~livereload_vs_browsersync) that favors Browser Sync. Browser Sync also has VS and VSCode plugins (if that matters).

**Comment by RLittlesII** [comment]
@MisinformedDNA I saw Browser Sync as an alternative, I did not find the nice comparison 😄 

My thoughts are to get the unit tests passing ASAP, so I can determine how to approach live reload.  There is a library called [Fleck](https://github.com/statianzo/Fleck) which is behind the `ReloadClient` that has been ported to netstandard2.0

I am game to go either direction, as I don't have experience with either myself.   I think @daveaglick will have to give us some guidance on direction here.

File: src/core/Wyam.Hosting/DefaultExtensionsMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    public class DefaultExtensionsMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IFileProvider _fileProvider;
+        private readonly DefaultExtensionsOptions _options;
+        private readonly ILogger _logger;
+
+        public DefaultExtensionsMiddleware(RequestDelegate next, IHostingEnvironment hostingEnv, IOptions<DefaultExtensionsOptions> options, ILoggerFactory loggerFactory)
+        {
+            if (next == null)
```
Comment (MisinformedDNA) [inline src/core/Wyam.Hosting/DefaultExtensionsMiddleware.cs:22]: You can do this as of C# 7: `_next = next ?? throw new ArgumentNullException();`

File: src/core/Wyam.Hosting/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        public Startup(IHostingEnvironment environment,
+            PreviewServerOptions previewServerOptions,
+            DefaultExtensionsOptions defaultExtensionsOptions)
+        {
+            _environment = environment;
+            _previewServerOptions = previewServerOptions;
+            _defaultExtensionsOptions = defaultExtensionsOptions;
+        }
+
+        public void Configure(IApplicationBuilder app)
```
Comment (MisinformedDNA) [inline src/core/Wyam.Hosting/Startup.cs]: This code is never called.

File: src/core/Wyam.Hosting/Owin/ExtensionlessUrlsExtensions.cs [hunk 3]
```diff
@@ -3,27 +3,25 @@
 using System.Linq;
```
Comment (MisinformedDNA) [inline src/core/Wyam.Hosting/Owin/ExtensionlessUrlsExtensions.cs]: I think this file can be removed in liu of the DefaultExtensions code.

File: src/core/Wyam.Hosting/Owin/ExtensionlessUrlsMiddleware.cs [hunk 4]
```diff
@@ -4,9 +4,11 @@
 using System.Runtime.Serialization.Formatters;
```
Comment (MisinformedDNA) [inline src/core/Wyam.Hosting/Owin/ExtensionlessUrlsMiddleware.cs]: I think this file can be removed in liu of the DefaultExtensions code

**Comment by daveaglick** [comment]
Thanks a ton for getting this going! I've merged it in after getting far enough along that the hosting tests pass (though I may have broken others in the process). Still some work to do on the actual server, but this got it far enough along that I've got a pretty good idea how to finish it off.

**Pull request merged**